`make gen` failed on arm64.

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:
`make gen` failed on arm64. here is the outputs:
```
root@vagrant:/app# make gen
go mod tidy
go: downloading github.com/prometheus/client_golang v1.20.3
go: downloading github.com/prometheus/common v0.59.1
go: downloading golang.org/x/oauth2 v0.22.0
Installed protoc version: 3.12.4
Installed protoc version does not match the desired version.
--2024-09-10 04:47:47--  https://github.com/protocolbuffers/protobuf/releases/download/v3.17.3/protoc-3.17.3-linux-x86_64.zip
Connecting to 192.168.1.169:7890... connected.
Proxy request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/23357588/403e2a00-c865-11eb-89ea-f8d8e896842a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240910%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240910T094747Z&X-Amz-Expires=300&X-Amz-Signature=27cc870b5abf09083f1aa4948181618ac0f5bcc6733aa78d886ab1f9304a15ea&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=23357588&response-content-disposition=attachment%3B%20filename%3Dprotoc-3.17.3-linux-x86_64.zip&response-content-type=application%2Foctet-stream [following]
--2024-09-10 04:47:48--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/23357588/403e2a00-c865-11eb-89ea-f8d8e896842a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240910%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240910T094747Z&X-Amz-Expires=300&X-Amz-Signature=27cc870b5abf09083f1aa4948181618ac0f5bcc6733aa78d886ab1f9304a15ea&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=23357588&response-content-disposition=attachment%3B%20filename%3Dprotoc-3.17.3-linux-x86_64.zip&response-content-type=application%2Foctet-stream
Connecting to 192.168.1.169:7890... connected.
Proxy request sent, awaiting response... 200 OK
Length: 1680819 (1.6M) [application/octet-stream]
Saving to: ‘protoc-3.17.3-linux-x86_64.zip’

protoc-3.17.3-linux-x86_64.zip                                         100%[===========================================================================================================================================================================>]   1.60M  2.06MB/s    in 0.8s

2024-09-10 04:47:49 (2.06 MB/s) - ‘protoc-3.17.3-linux-x86_64.zip’ saved [1680819/1680819]

Archive:  protoc-3.17.3-linux-x86_64.zip
   creating: protoc-3.17.3/include/
   creating: protoc-3.17.3/include/google/
   creating: protoc-3.17.3/include/google/protobuf/
  inflating: protoc-3.17.3/include/google/protobuf/empty.proto
  inflating: protoc-3.17.3/include/google/protobuf/wrappers.proto
  inflating: protoc-3.17.3/include/google/protobuf/type.proto
  inflating: protoc-3.17.3/include/google/protobuf/timestamp.proto
  inflating: protoc-3.17.3/include/google/protobuf/descriptor.proto
   creating: protoc-3.17.3/include/google/protobuf/compiler/
  inflating: protoc-3.17.3/include/google/protobuf/compiler/plugin.proto
  inflating: protoc-3.17.3/include/google/protobuf/source_context.proto
  inflating: protoc-3.17.3/include/google/protobuf/api.proto
  inflating: protoc-3.17.3/include/google/protobuf/struct.proto
  inflating: protoc-3.17.3/include/google/protobuf/duration.proto
  inflating: protoc-3.17.3/include/google/protobuf/any.proto
  inflating: protoc-3.17.3/include/google/protobuf/field_mask.proto
   creating: protoc-3.17.3/bin/
  inflating: protoc-3.17.3/bin/protoc
  inflating: protoc-3.17.3/readme.txt
protoc already installed
protoc-c already installed
protoc-gen-go already installed
   PROTO   api/v2-c
Generate proto files in docker
   PROTO   api/v2
/usr/local/bin/protoc: 1: cannot create
                                        ~
                                         ~ddQ�td/lib64/ld-linux-x86-64.so.2GNUGNU����ۣ��@�
��ԛ[x�#@��������8�fe�I�Y@�}u˪�"��:���J��oK�kqyq�
�o��wH�: Directory nonexistentZ��SEN=�6�_
/usr/local/bin/protoc: 1: ELF: not found
/usr/local/bin/protoc: 4: Syntax error: "(" unexpected
make[1]: *** [Makefile:41: gen-proto] Error 2
make: *** [Makefile:99: gen-proto] Error 2
```
**What you expected to happen**:
`make gen` succeeds.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
protoc is downloaded form a x86 srouce url. need a arm version.

**Environment**:
- Kmesh version:
- Others: ubuntu arm64

--- a/hack/install-proto.sh
+++ b/hack/install-proto.sh
@@ -11,31 +11,55 @@ function install_tool() {
     fi
 }
 
+function install_protoc_binary() {
+    local version="$1"
+    local arch_suffix="$2"
+    local url="$3"
+
+    wget "$url" -O "protoc-${version}-linux-${arch_suffix}.zip"
+    unzip "protoc-${version}-linux-${arch_suffix}.zip" -d "protoc-${version}"
+    sudo mv "protoc-${version}/bin/protoc" /usr/local/bin/
+    rm -rf "protoc-${version}-linux-${arch_suffix}.zip" "protoc-${version}"
+}
+
 function install_protoc() {
-    desired_version="3.17.3"
+    local arch
+    local arch_suffix
+    local download_url
+    local installed_version
+    local desired_version="3.17.3"
+
+    # check the architecture
+    arch=$(uname -m)
+    case "$arch" in
+        x86_64)
+            arch_suffix="x86_64"
+            ;;
+        aarch64)
+            arch_suffix="aarch_64"
+            ;;
+        *)
+            echo "Unsupported architecture: $arch"
+            exit 1
+            ;;
+    esac
+
+    download_url="https://github.com/protocolbuffers/protobuf/releases/download/v${desired_version}/protoc-${desired_version}-linux-${arch_suffix}.zip"
+
     # Check if protoc is installed
     if command -v protoc >/dev/null 2>&1; then
-        # Get the installed version
         installed_version=$(protoc --version | awk '{print $2}')
         echo "Installed protoc version: $installed_version"
-        # Compare the installed version with the desired version
-        if [[ "$installed_version" == "$desired_version" ]]; then
-            echo "Installed protoc version matches the desired version."
-        else
-            echo "Installed protoc version does not match the desired version."
-            # install protoc
-            wget https://github.com/protocolbuffers/protobuf/releases/download/v3.17.3/protoc-3.17.3-linux-x86_64.zip
-            unzip protoc-3.17.3-linux-x86_64.zip -d protoc-3.17.3
-            sudo mv protoc-3.17.3/bin/protoc /usr/local/bin/
-            rm -rf protoc-3.17.3-linux-x86_64.zip protoc-3.17.3
-        fi
     else
-        echo "protoc is not installed"
-        # install protoc
-        wget https://github.com/protocolbuffers/protobuf/releases/download/v3.17.3/protoc-3.17.3-linux-x86_64.zip
-        unzip protoc-3.17.3-linux-x86_64.zip -d protoc-3.17.3
-        sudo mv protoc-3.17.3/bin/protoc /usr/local/bin/
-        rm -rf protoc-3.17.3-linux-x86_64.zip protoc-3.17.3
+        installed_version="none"
+        echo "protoc is not installed."
+    fi
+
+    if [[ "$installed_version" != "$desired_version" ]]; then
+        echo "Installing protoc version ${desired_version}..."
+        install_protoc_binary "$desired_version" "$arch_suffix" "$download_url"
+    else
+        echo "protoc is up-to-date."
     fi
 
     tools=()

